modules/xfd.ts

```diff
@@ -0,0 +1,2218 @@
+				params.initialContrib = null; // disable initial contributor logging in userspace log
+				statusElement.warn('You (' + usernameOrTarget + ') created this page; skipping user notification');
+				return def.resolve();
+			}
+		}
+
+		var usertalkpage = new Morebits.wiki.page(notifyPageTitle, statusElement);
+		usertalkpage.setAppendText('\n\n' + this.getNotifyText());
+		usertalkpage.setEditSummary(this.getNotifyEditSummary());
+		usertalkpage.setChangeTags(Twinkle.changeTags);
+		usertalkpage.setCreateOption('recreate');
+		// Different pref for RfD target notifications: XXX: handle this better!
+		if (params.venue === 'rfd' && targetNS !== 3) {
+			usertalkpage.setWatchlist(Twinkle.getPref('xfdWatchRelated'));
+		} else {
+			usertalkpage.setWatchlist(Twinkle.getPref('xfdWatchUser'));
+		}
+		usertalkpage.setFollowRedirect(true, false);
+		usertalkpage.append(def.resolve, function onNotifyError() {
+			// if user could not be notified, null this out for correct userspace logging
+			params.initialContrib = null;
+			def.resolve();
+		});
+		return def;
+	}
+
+	// Overridden for all venues except FFD and RFD
+	getNotifyText(): string {
+		return `{{subst:${this.params.venue} notice|1=${Morebits.pageNameNorm}}} ~~~~`;
+	}
+
+	// Not overridden for any venue
+	getNotifyEditSummary(): string {
+		return 'Notification: [[' + this.params.discussionpage + '|listing]] of [[:' +
+		Morebits.pageNameNorm + ']] at [[WP:' + this.getFieldsetLabel() + ']].';
+	}
+
+	notifyCreator(): JQuery.Promise<void> {
+		if (!this.params.notifycreator) {
+			this.params.intialContrib = null;
+			return $.Deferred().resolve();
+		}
+		return this.notifyTalkPage(this.params.initialContrib);
+	}
+
+	// Should be called after notifyTalkPage() which may unset this.params.intialContrib
+	addToLog() {
+		let params = this.params;
+
+		if (!Twinkle.getPref('logXfdNominations') ||
+			Twinkle.getPref('noLogOnXfdNomination').indexOf(params.venue) !== -1) {
+			return $.Deferred().resolve();
+		}
+
+		var usl = new Morebits.userspaceLogger(Twinkle.getPref('xfdLogPageName'));// , 'Adding entry to userspace log');
+
+		usl.initialText =
+			"This is a log of all [[WP:XFD|deletion discussion]] nominations made by this user using [[WP:TW|Twinkle]]'s XfD module.\n\n" +
+			'If you no longer wish to keep this log, you can turn it off using the [[Wikipedia:Twinkle/Preferences|preferences panel]], and ' +
+			'nominate this page for speedy deletion under [[WP:CSD#U1|CSD U1]].' +
+			(Morebits.userIsSysop ? '\n\nThis log does not track XfD-related deletions made using Twinkle.' : '');
+
+		usl.changeTags = Twinkle.changeTags;
+		return usl.log(this.getUserspaceLoggingText(), this.getUserspaceLoggingEditSummary());
+	}
+
+	getUserspaceLoggingEditSummary() {
+		return 'Logging ' + utils.toTLACase(this.params.venue) + ' nomination of [[:' + Morebits.pageNameNorm + ']].';
+	}
+
+	getUserspaceLoggingText(): string {
+		let params = this.params;
```

Comment:
definitely not too familiar with typescript, but is there a functional difference here between `let` and `var`? If not, suggest being consistent one way or the other

Reply:
I used `let` in all new code, and retained `var`s from the code copied from the old version. Maybe just do a find-and-replace of all `var`s with `let`?